internal/extsvc/bitbucketserver/events.go

```diff
@@ -44,8 +56,25 @@ type PushEvent struct {
 	Repository Repo `json:"repository"`
 }
 
+type CustomTime time.Time
+
+// UnmarshalJSON we create a custom unmarshal func to handle the custom time format
+// from Bitbucket
+func (ct *CustomTime) UnmarshalJSON(b []byte) error {
+	// The date returned by the Bitbucket webhook is a string in the format:
+	// "2016-07-19T12:34:56+0000". We have to strip out the extraneous quote
+	// before parsing the string.
+	s := strings.Trim(string(b), "\"")
+	t, err := time.Parse("2006-01-02T15:04:05+0000", s)
```

Comment:
```suggestion
	t, err := time.Parse("2006-01-02T15:04:05-0700", s)
```

The last bit there is timezone information, which you can represent with `-0700` according to the `time` package